pymeasure/instruments/hp/hp8753e.py

```diff
@@ -0,0 +1,394 @@
+
+
+class HP8753E(Instrument):
+    """Represents the HP8753E Vector Network Analyzer
+    and provides a high-level interface for taking scans of the
+    scattering parameters.
+
+    Template copied from Agilent 8722 class in pymeasure
+    """
+
+    SCAN_POINT_VALUES = [3, 11, 21, 26, 51, 101, 201, 401, 801, 1601]
+    SCATTERING_PARAMETERS = ["S11", "S12", "S21", "S22"]
+    S11, S12, S21, S22 = SCATTERING_PARAMETERS
+
+    start_frequency = Instrument.control(
+        "STAR?",
+        "STAR %e Hz",
+        """ A floating point property that represents the start frequency
+        in Hz. This property can be set.
+        """,
+    )
+    stop_frequency = Instrument.control(
+        "STOP?",
+        "STOP %e Hz",
+        """ A floating point property that represents the stop frequency
+        in Hz. This property can be set.
+        """,
+    )
+    sweep_time = Instrument.control(
+        "SWET?",
+        "SWET%.2e",
+        """ A floating point property that represents the sweep time
+        in seconds. This property can be set.
+        """,
+    )
+    averages = Instrument.control(
+        "AVERFACT?",
+        "AVERFACT%d",
+        """ An integer representing the number of averages to take. Note that
+        averaging must be enabled for this to take effect. This property can be set.
+        """,
+        cast=lambda x: int(float(x)),  # need float() to convert scientific notation in strings
+    )
+    averaging_enabled = Instrument.control(
+        "AVERO?",
+        "AVERO%d",
+        """ A bool that indicates whether or not averaging is enabled. This property
+        can be set.""",
+        cast=bool,
+    )
+
+    correction_enabled = Instrument.control(
+        "CORR?",
+        "CORR%d",
+        """ A bool that indicates whether or not correction is enabled. This property
+        can be set.""",
+        cast=bool,
+    )
+
+    power = Instrument.control(
+        "POWE?",
+        "POWE%.2e",
+        """ A float that can be set from -70 to +10dBm to control output power from the VNA ports. 
+        This property can be set.""",
+        cast=lambda x: float(x),
+    )
+
+    def __init__(self, adapter, name="Hewlett Packard 8753E Vector Network Analyzer", **kwargs):
+        # this should probably be in __new__
+        super().__init__(adapter, name, **kwargs)
+        if isinstance(self.adapter, PrologixAdapter):
+            self.adapter.auto = 1
+            self.adapter.connection.query_delay = 0
+            self.adapter.gpib_read_timeout = 500
+            self.adapter.connection.timeout = 700
+            self.adapter.eoi = 1
+            self.adapter.eos = "\n"
+            self.adapter.connection.read_termination = self.adapter.eos
+            self.adapter.connection.read_termination = "\r\n"
+            self.adapter.write("++eot_enable 0")
+            self.adapter.flush_read_buffer()
+            self.write("FORM4")
+            self.adapter.flush_read_buffer()
```

Comment:
You asked, how to define the connection settings:

- First, the driver (this class) should define the communication expected from the computer without any Adapter in between, therefore you should define the GPIB parameters in the `super().__init__` call, see the [documentation](https://pymeasure.readthedocs.io/en/latest/dev/adding_instruments/instrument.html#defining-default-connection-settings).
- If you use a non default adapter (for example Prologix instead of the default VISAAdapter), you have to set up that adapter outside of this file, that is, in your personal code.